### ISSUE TITLE
Should `result.exitCode` be `undefined`?

### ISSUE THREAD

**Issue by ehmicky** [sindresorhus/nano-spawn#42]
Right now, the returned promise either:
  - Resolves with a `result` plain object, where `result.exitCode` is `0`
  - Rejects with an error, where `error.exitCode` is either `undefined` (e.g. during a `SIGTERM`) or non-`0`

Since `result.exitCode` is always `0`, is it useful? Should it be `undefined` instead then?

For example, `signalName` is only defined on `error`. Similarly, `exitCode` is possibly an information that is meant for errors.

With Execa, we define `error` as an extension of the `result` object. However, `nano-spawn` does not have the `reject` option, so it might make less sense.

In other words, should we:
  1. Keep things as is?
  2. Or define only `error.exitCode` (like we only define `error.signalName`), meaning `result.exitCode` would be always `undefined`?

What are your thoughts on this?

**Comment by sindresorhus** [comment]
Definitely `2`
